docs/translating.md

```diff
@@ -76,25 +76,111 @@ To ensure we always internationalize our JSON errors messages, the
 Zulip linter (`tools/lint-all`) checks for correct usage.
 
 ## Frontend Translations
-The first step in translating the frontend is to create the translation
-files using `python manage makemessages`. This command will create
-translation files under `static/locale`, the location can be changed by
-passing an argument to the command, however make sure that the location is
-publicly accessible since these files are loaded through XHR in the
-frontend which will only work with publicly accessible resources.
+Zulip uses the [i18next][] library for the frontend translations. To make
+the frontend translations work, the first step is to create the JSON
+formatted [resource][] files using the `python manage makemessages`
+command. This command will create a resource file called `translations.json`
+for every language under `static/locale`. The location can be changed by
+passing an argument to the command (see the help for the command for
+further details). However, make sure that the location is publicly
+accessible since these files are loaded through XHR in the frontend
+which will only work with publicly accessible resources.
+
+Note: `makemessages` command is idempotent in a way that it will only
+delete singular keys in the resource file when they are not present in
+Zulip code anymore; it will only delete plural keys, see below for the
+documentation on plural translations, when the corresponding singular
+key is absent, and it will not override the value of a singular key if
+that value contains a translation.
```

Comment:
I think you're saying that `makemessages` deletes singular keys if they aren't used in source code anymore, right?

Here's a suggested revision, to fix the punctuation and grammar:

Note: the `makemessages` command is idempotent in that:
* it will only delete singular keys in the resource file when they are no longer used in Zulip code
* it will only delete plural keys (see below for the documentation on plural translations) when the corresponding singular key is absent
* it will not override the value of a singular key if that value contains a translation